### PR TITLE
Add build-only for quickstart builds.

### PR DESCRIPTION
**Pull request by granluo**
Add `mock-GoogleService-Info.plist` for Quickstarts that need build only.

### CODE REVIEW

**Review by paulb777 — COMMENTED** [review]
I'm not sure the scripts need to change. 

In order to solve the problem of being able to develop the quickstarts and add and remove files to them, the GoogleService-Info.plist needs to already be in the project so that it is not cumbersome to remove it between testing and making commits.

Therefore my proposal would be to 

1. copy mock-GoogleService-Info.plist to database/GoogleService-Info.plist
2. Add it to the database project
3. Verify the project builds and passes CI
4. Update all the other quickstart's similarly
5. Remove the else clause in prereq_core.sh
6. Stop running info_script.rb
7. Update the README's to say replace the GoogleService-Info.plist instead of adding it

cc: @morganchen12 for any comments or concerns about this plan

**Comment by morganchen12** [comment]
We'll need to update scripts to both remove the mock plist and insert the test plist from `Secrets` when testing with secrets available but otherwise lgtm

File: database/DatabaseExample.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -37,6 +37,9 @@
 		10B872C61CDC14C800C7D214 /* MyTopPostsViewController.swift in Sources */ = {isa = PBXBuildFile; fileRef = 10B872C51CDC14C800C7D214 /* MyTopPostsViewController.swift */; };
 		10F2262F1CDBEE7C005205F2 /* SignInViewController.m in Sources */ = {isa = PBXBuildFile; fileRef = 10F2262E1CDBEE7C005205F2 /* SignInViewController.m */; };
 		10F226321CDBEE95005205F2 /* Post.m in Sources */ = {isa = PBXBuildFile; fileRef = 10F226311CDBEE95005205F2 /* Post.m */; };
+		38842A419DD14AFDD4D05807 /* Pods_DatabaseExample.framework in Frameworks */ = {isa = PBXBuildFile; fileRef = B2516808CC7DCDC39AA25B9D /* Pods_DatabaseExample.framework */; };
```
Comment (morganchen12) [inline database/DatabaseExample.xcodeproj/project.pbxproj:40]: Run pod deintegrate to remove these changes.

**Comment by granluo** [comment]
The mock-GoogleService-Info.plist will be forced added since it is in the `.ignore`.
CI scripts need to be updated for testing with secrets.

File: database/DatabaseExample.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -267,6 +273,13 @@
 			name = ViewControllers;
 			sourceTree = "<group>";
 		};
+		244A75058F265EBB9D457A95 /* Pods */ = {
```
Comment (paulb777) [inline database/DatabaseExample.xcodeproj/project.pbxproj:276]: Remove Pods references in this file.  This can be done manually or from the Xcode UI

**Review by morganchen12 — COMMENTED** [review]
LGTM, but deferring to Paul for approval.

**Review by paulb777 — APPROVED** [review]
Need a similar change to https://github.com/firebase/quickstart-ios/blob/master/database/DatabaseExampleSwiftUI/DatabaseExample/DatabaseExample.xcodeproj/project.pbxproj to help @judyliu14 - but that can be a separate PR.

Also, the firebase-ios-sdk may need an follow-on PR to also not update the project.pbxproj

**Comment by granluo** [comment]
> Need a similar change to https://github.com/firebase/quickstart-ios/blob/master/database/DatabaseExampleSwiftUI/DatabaseExample/DatabaseExample.xcodeproj/project.pbxproj to help @judyliu14 - but that can be a separate PR.
> 
> Also, the firebase-ios-sdk may need an follow-on PR to also not update the project.pbxproj

The change will probably break the [database workflow in the sdk repo](https://github.com/firebase/firebase-ios-sdk/blob/master/.github/workflows/database.yml#L91-L94) due to [an extra `GoogleService-Info.plist` was added]( https://github.com/firebase/firebase-ios-sdk/blob/master/scripts/setup_quickstart.sh#L82). To resolve this, will
1.  Update the change to the rest of Quickstarts
2. Remove [plist adding line](https://github.com/firebase/firebase-ios-sdk/blob/master/scripts/setup_quickstart.sh#L82) when a quickstart is setup in the SDK repo.
3. Ensure [plist from secrets](https://github.com/firebase/firebase-ios-sdk/blob/master/.github/workflows/database.yml#L93-L94) can override the dummy plist

**Comment by granluo** [comment]
Will submit the PR and work on the rest of Quickstarts and also adjust the workflows in the SDK repo.

**Pull request merged**